src/psyclone/domain/common/transformations/kernel_module_inline_trans.py

```diff
@@ -250,68 +296,127 @@ def _prepare_code_to_inline(code_to_inline):
                     symbol.interface.container_symbol = \
                         code_to_inline.symbol_table.lookup(module_symbol.name)
 
-    def apply(self, node, options=None):
+    @staticmethod
+    def _get_psyir_to_inline(node):
+        '''
+        Wrapper that gets the name and PSyIR of the routine or kernel
+        corresponding to the call described by `node`.
+
+        :param node: the Call or CodedKern to resolve.
+        :type node: :py:class:`psyclone.psyir.nodes.Call` |
+                    :py:class:`psyclone.psyGen.CodedKern`
+
+        :returns: the name of the routine as seen by the caller and the
+                  PSyIR of the routine implementation.
+        :rtype: Tuple(str, :py:class:`psyclone.psyir.nodes.Call`)
+
+        '''
+        if isinstance(node, CodedKern):
+            # We have a call to a Kernel in a PSyKAl API.
+            code_to_inline = node.get_kernel_schedule()
+            caller_name = node.name.lower()
+        else:
+            # We have a generic routine call.
+            code_to_inline = node.routine.get_routine()
+            caller_name = node.routine.name.lower()
+        return (caller_name, code_to_inline)
+
+    def apply(self, node, options=()):
         ''' Bring the kernel subroutine into this Container.
 
         :param node: the kernel to module-inline.
         :type node: :py:class:`psyclone.psyGen.CodedKern`
         :param options: a dictionary with options for transformations.
-        :type options: Optional[Dict[str, Any]]
+        :type options: Optional[Iterable]
```

Comment:
Unfortunately I don't think the tuple/Iterable trick works here to remove the:
```
        if not options:
            options = {}
```
This worked when we only iterate the list/dict on the body of the method, in general the options argument is something that we access by name and therefore we use the Mapping interface (https://docs.python.org/3/glossary.html#term-mapping), but I don't think we have an immutable object that implements the Mapping interface that we could provide as default argument (unless we import some non-standard class, but I don't think its worth it)

Reply:
Oh yes. I've just reverted this change.